MiniSim/Service/DeviceService.swift

```diff
@@ -263,14 +308,96 @@ extension DeviceService {
 
   static func getIOSDevices() throws -> [Device] {
     Thread.assertBackgroundThread()
+    let simulators = try getIOSSimulators()
+    let devices = try getIOSPhysicalDevices()
+    return simulators + devices
+  }
+
+  static var currentMachineUUID: String? {
+    guard let output = try? shellOut(
+      to: ProcessPaths.systemProfiler.rawValue,
+      arguments: ["SPHardwareDataType"]
+    ) else { return nil }
+
+    let uuidIdx = 1
+    for line in output.components(separatedBy: "\n") {
+      guard let match = line.match("Hardware UUID: ([\\S]+)").first else {
+        continue
+      }
+
+      if match.count >= 2 {
+        return match[uuidIdx]
+      }
+    }
+
+    return nil
+  }
+
+  struct DeviceCtlJson: Codable {
+    struct Info: Codable {
+      var outcome: String
+    }
+
+    struct DeviceProperties: Codable {
+      let name: String
+      let osVersionNumber: String
+    }
 
+    struct HardwareProperties: Codable {
+      let udid: String
+    }
+
+    struct ConnectionProperties: Codable {
+      let tunnelState: String
+    }
+
+    struct Device: Codable {
+      let deviceProperties: DeviceProperties
+      let hardwareProperties: HardwareProperties
+      let connectionProperties: ConnectionProperties
+    }
+
+    struct Result: Codable {
+      let devices: [Device]
+    }
+
+    let info: Info
+    let result: Result
+  }
```

Comment:
I think this can be moved to DeviceParsers (the new class I've added) we can overload the file buffer there and do all decoding

Reply:
How would you do the overloading?

Reply:
@okwasniewski can you check my pushed changes if that matches what you mean?